wikirace.go

```diff
@@ -0,0 +1,57 @@
+package main
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+)
+
+type articleInfo struct {
+	ID    int    `json:"id"`
+	Title string `json:"title"`
+}
+
+type queryResult struct {
+	Random []articleInfo `json:"random"`
+}
+
+type wikiResult struct {
+	BatchComplete string      `json:"batchComplete"`
+	Query         queryResult `json:"query"`
+}
+
+func initWikiRace(ctx *context) {
+	m := ctx.Message
+	s := ctx.Session
+
+	resp, err := http.Get("https://en.wikipedia.org/w/api.php?action=query&format=json&list=random&rnnamespace=0&rnlimit=2")
+	if err != nil {
+		log.Error(err)
+		s.ChannelMessageSend(m.ChannelID, "Hmm... I seem to have run into an issue... Try again later?")
+		return
+	}
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		log.Error(err)
+		s.ChannelMessageSend(m.ChannelID, "Hmm... I seem to have run into an issue... Try again later?")
+		return
+	}
+
+	var search wikiResult
+	err = json.Unmarshal(body, &search)
+	if err != nil {
+		log.Error(err)
+		s.ChannelMessageSend(m.ChannelID, "Hmm... I seem to have run into an issue... Try again later?")
+		return
+	}
+
+	articles := search.Query.Random[:2]
+
+	// @carson is there a better way of doing this?
```

Comment:
I'd use a string builder probably (`strings.Builder` I believe), but this isn't wrong. Maybe in a future version, sending the start and end as an embed would work better.